src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java

```diff
@@ -16,17 +17,157 @@ public abstract class SparkMaxMotor implements IMotor {
 	protected final SparkMaxWrapper motor;
 	private final String logPath;
 	private final ConnectedInputAutoLogged connectedInput;
+	private SparkBase.Warnings warnings;
+	private SparkBase.Faults faults;
 
 	public SparkMaxMotor(String logPath, SparkMaxWrapper motor) {
 		this.logPath = logPath;
 		this.motor = motor;
+		this.warnings = motor.getWarnings();
+		this.faults = motor.getFaults();
 
 		this.connectedInput = new ConnectedInputAutoLogged();
 		connectedInput.connected = true;
 
 		AlertManager.addAlert(new PeriodicAlert(Alert.AlertType.ERROR, logPath + "disconnectedAt", () -> !isConnected()));
 	}
 
+	public void createFaultAlerts() {
+		//@formatter:off
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "OtherErrorAt",
+						() -> faults.other
+				)
+		);
+
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "MotorTypeMismatchAt",
+						() -> faults.motorType
+				)
+		);
+
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "ConnectedSensorFaultAt",
+						() -> faults.sensor
+				)
+		);
+
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "CANFatalFaultAt",
+						() -> faults.can
+				)
+		);
+
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "OverHeatingAt",
+						() -> faults.temperature
+				)
+		);
+
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "GateDriveCircuitryFaultAt",
+						() -> faults.gateDriver
+				)
+		);
+
+		AlertManager.addAlert(
+				new PeriodicAlert(
+						Alert.AlertType.ERROR,
+						logPath + "EscEepromFaultAt",
```

Comment:
Esc? like the ftc competition?

Reply:
Do you prefer Electronic Speed Controller Electrically Erasable Programmable Read-Only Memory?
Genuine question